bot/cogs/match.py

```diff
@@ -301,7 +302,11 @@ def all_ready(reaction, user):
                 return False
 
         try:
-            await self.bot.wait_for('reaction_add', timeout=60.0, check=all_ready)
+            if self.pending_task is not None:
+                self.pending_task.close()
+
+            self.pending_task = self.bot.wait_for('reaction_add', timeout=60.0, check=all_ready)
+            await self.pending_task
```

Comment:
Would need to change this section to use a dictionary instead. Can check if the `guild` is in the dictionary keys and add the task for that `guild` if not.